src/index.js

```diff
@@ -62,6 +85,59 @@ function ContextProvider(props) {
 	return cloneElement(children, rest);
 }
 
+function cloneElementsToShadowRoot(shadowRoot, elements) {
+	elements.forEach((el) => shadowRoot.appendChild(el.cloneNode(true)));
+}
+
+function getAllStyles(target, selector, filter) {
+	const elements = Array.prototype.slice.call(
+		target.querySelectorAll(selector)
+	);
+
+	return filter ? elements.filter(filter) : elements;
+}
+
+const beginInjectingGlobalStyles = (shadowRootRef, injectGlobalStyles) => {
+	cloneElementsToShadowRoot(
+		shadowRootRef,
+		getAllStyles(
+			injectGlobalStyles.target,
+			injectGlobalStyles.selector,
+			injectGlobalStyles.filter
+		)
+	);
+
+	return observeStyleChanges(
+		(elements) => {
+			cloneElementsToShadowRoot(shadowRootRef, elements);
+		},
+		injectGlobalStyles.target,
+		injectGlobalStyles.selector,
+		injectGlobalStyles.filter,
+		injectGlobalStyles.observeOptions
+	);
+};
+
+function observeStyleChanges(
+	callback,
+	target,
+	selector,
+	filter,
+	observeOptions
+) {
+	return new MutationObserver((mutations, observer) => {
+		mutations.forEach((mutation) => {
+			const matchedElements = Array.prototype.slice
```

Comment:
Worth checking: If we inline `cloneElementsToShadowRoot` we can loop directly over the `NodeList`, thereby skipping the additional iteration over elements and saving an array allocation in the process.